loopchain/peer/block_manager.py

```diff
@@ -809,10 +826,14 @@ def verify_confirm_info(self, unconfirmed_block: Block):
             block_verifier.verify(unconfirmed_block,
                                   self.blockchain.last_block,
                                   self.blockchain,
-                                  ExternalAddress.fromhex_address(
-                                      self.blockchain.get_expected_generator(unconfirmed_block.header.peer_id)
-                                  ),
+                                  self.blockchain.get_expected_generator(unconfirmed_block.header.peer_id),
                                   reps_getter=reps_getter)
+        except NotInReps as e:
+            util.logger.debug(f"in _vote Not In Reps({e}) state({self.__channel_service.state_machine.state})")
+            self.__channel_service.switch_role()
```

Comment:
Currently, `switch_role` is intended only when citizen node switches to community node. Is it okay to call it from Community node?

Reply:
It will support that way and there is no problem at present.